Iris/IrisCompiler/CompilerContext.cs

```diff
@@ -112,12 +111,18 @@ public void AddIntrinsics()
             {
                 ImportGlobalField(fp, "$.emptystr", mscorlib, "System.String", "Empty");
                 ImportMethod(fp, "concat", mscorlib, "System.String", "Concat", false, tString, new IrisType[] { tString, tString });
-                ImportMethod(fp, "readln", mscorlib, "System.Console", "ReadLine", false, tString, noParams);
                 ImportMethod(fp, "str", mscorlib, "System.Int32", "ToString", true, tString, noParams);
                 ImportMethod(fp, "strcmp", mscorlib, "System.String", "Compare", false, tInt, new IrisType[] { tString, tString });
-                ImportMethod(fp, "writeln", mscorlib, "System.Console", "WriteLine", false, tVoid, new IrisType[] { tString });
+
+                ImportedModule consoleLib = ReferenceConsoleLib();
+                if (consoleLib != null)
+                {
+                    ImportMethod(fp, "readln", consoleLib, "System.Console", "ReadLine", false, tString, noParams);
+                    ImportMethod(fp, "writeln", consoleLib, "System.Console", "WriteLine", false, tVoid, new IrisType[] { tString });
+                }
```

Comment:
>} [](start = 16, length = 1)

`else` Import `readln` and `writeln` from mscorlib.

Reply:
No. If somehow that method returns false then it will not be in mscorlib.

Reply:
Ah, I understand.  ReferenceConsoleLib will return mscorlib on Desktop CLR.  It wasn't clear to me from the name.

---
In reply to: [562862041](https://github.com/microsoft/ConcordExtensibilitySamples/pull/65#discussion_r562862041) [](ancestors = 562862041)